spec/controllers/admin/booths_controller_spec.rb

```diff
@@ -0,0 +1,152 @@
+require 'spec_helper'
+
+describe Admin::BoothsController do
+
+  let(:admin) { create(:admin) }
+  let(:conference) { create(:conference) }
+  let(:booth) { create(:booth, conference: conference) }
+  let(:admin) { create(:admin) }
+
+  context 'not logged in user' do
+
+    describe 'GET index' do
+      it 'does not render admin/booths#index' do
+        get :index, conference_id: conference.short_title
+        expect(response).to redirect_to(user_session_path)
+      end
+    end
+
+    # describe 'GET show' do
+    #   it 'does not render admin/booths#show' do
+    #     get :show, conference_id: conference.short_title
+    #     expect(response).to redirect_to(user_session_path)
+    #   end
+    # end
+  end
+
+  context 'user is admin' do
+    before :each do
+      sign_in admin
+    end
+
+    describe 'GET index' do
+      before { get :index, conference_id: conference.short_title }
+
+      it 'assigns attributes for booths' do
+        expect(assigns(:booths)).to eq([booth])
+      end
+
+      it 'renders index template' do
+        expect(response).to render_template('index')
+      end
+    end
+
+    describe 'GET new' do
+      before { get :new, conference_id: conference.short_title }
+
+      it 'assigns attributes for booths' do
+        expect(assigns(:booth)).to be_a_new(Booth)
+      end
+
+      it 'renders new template' do
+        expect(response).to render_template('new')
+      end
+    end
+
+    describe 'POST #create' do
+      # context 'successfully created' do
+      #   it 'creates a new booth' do
+      #     expected = expect do
+      #       post :create, booth: attributes_for(:booth), conference_id: conference.short_title
+      #     end
+      #     expected.to change { Booth.count }.by(1)
+      #   end
+      #
+      #   it 'redirects to admin booth index' do
+      #     post :create, booth: attributes_for(:booth), conference_id: conference.short_title
+      #     expect(response).to redirect_to(admin_conference_booths_path)
```

Comment:
When this fails:

`  3) Admin::BoothsController user is admin POST #create successfully created redirects to admin booth index
     Failure/Error: expect(response).to redirect_to(admin_conference_booths_path)
       Expected response to be a <redirect>, but was <200>
     # ./spec/controllers/admin/booths_controller_spec.rb:67:in `block (5 levels) in <top (required)>'
`

So it does not redirect, but it has a success status code, so we should be thinking it has rendered something.